pkg/filters/proxy/pool.go

```diff
@@ -559,6 +559,10 @@ func (sp *ServerPool) doHandle(stdctx stdcontext.Context, spCtx *serverPoolConte
 		return serverPoolError{http.StatusInternalServerError, resultInternalError}
 	}
 
+	if clb, ok := sp.LoadBalancer().(CustomizableLoadBalancer); ok {
+		clb.Customize(svr, spCtx.resp)
+	}
+
```

Comment:
please add a new function to the `LoadBalancer` interface instead of using a new interface, otherwise, developers may adding more and more customized interfaces, and we will have to type assert every interface here.